File: projecttests/go/tests/helloworld/worker.go
package helloworld

import (
	"github.com/temporalio/omes/projecttests/go/harness"
	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
)

func workerMain(config *harness.WorkerConfig) error {
	c, err := client.Dial(config.ConnectionOptions)
	if err != nil {
		return err
	}
	defer c.Close()

	w := worker.New(c, config.TaskQueue, worker.Options{})
	w.RegisterWorkflow(Workflow)
	return w.Run(worker.InterruptCh())
}

File: workflowtests/python/tests/simple_test/simple_test/worker.py
from omes_starter import WorkerConfig
from temporalio.client import Client
from temporalio.worker import Worker

from .workflows import SimpleWorkflow


async def worker_main(config: WorkerConfig) -> Worker:
    """Configure and return the worker."""
    client = await Client.connect(config.server_address, **config.connect_kwargs())
    return Worker(
        client,
        task_queue=config.task_queue,
        workflows=[SimpleWorkflow],
    )
